### DISCUSSION TITLE
[Q-Select] drop down width cannot be set

### DISCUSSION THREAD

**Discussion by moritzhuether** [quasarframework/quasar#16630] [General - Components / Directives / etc]
Currently, the width of the dropdown container of `q-select` sets its `min-width` to the width of the input. However, when a long item is loaded into the dropdown its width changes according to the width of the item. I want the container to be the same size as the input, regardless of item length or a potential scroll bar.

#12678 and #12617 proposed solutions, but they only work if there is no scroll bar as seen in the attached image.

![image](https://github.com/quasarframework/quasar/assets/51322048/32e0fa32-2aa4-47a3-b119-3fe3e9c90e52)

Has anyone an idea how to fix this?

**Comment by metalsadman** [comment]
@moritzhuether try playing with computing the width something like this https://codepen.io/metalsadman/pen/PoOvqBB

**Reply by moritzhuether** [reply]
Thank you @metalsadman.

This solution has the issue of always having a 15px gap to the right side, regardless of the visibility of the scrollbar. I added a negative margin of 15px so that at least the hover/selection background color would reach the side.

Is there any way to get access to the container that the q-select is rendering for the options? Wouldn't it be possible to make this container a slot?

**Reply by metalsadman** [reply]
@moritzhuether updated the codepen https://codepen.io/metalsadman/pen/PoOvqBB?editors=1010
taken from here on getting the scrollbar width https://stackoverflow.com/a/40568748
